rpc-state-reader/src/blockifier_state_reader.rs

```diff
@@ -404,6 +244,26 @@ pub fn execute_tx_configurable(
     Ok((blockifier_exec_info, trace, receipt))
 }
 
+pub fn build_cached_state(network: &str, current_block_number: u64) -> CachedState<RpcStateReader> {
+    let previous_block_number = BlockNumber(current_block_number - 1);
+    let rpc_chain = parse_network(network);
+    let rpc_reader = RpcStateReader(
+        RpcState::new_rpc(rpc_chain, previous_block_number.into())
+            .expect("failed to create state reader"),
+    );
+
+    CachedState::new(rpc_reader)
+}
```

Comment:
It should create the cache for the specified block number, the `block_number-1` is done before calling the function.
```suggestion
pub fn build_cached_state(network: &str, block_number: u64) -> CachedState<RpcStateReader> {
    let block_number = BlockNumber(block_number);
    let rpc_chain = parse_network(network);
    let rpc_reader = RpcStateReader(
        RpcState::new_rpc(rpc_chain, block_number.into())
            .expect("failed to create state reader"),
    );

    CachedState::new(rpc_reader)
}
```